### DISCUSSION TITLE
Check For Blank Query Result

### DISCUSSION THREAD

**Discussion by JJWest12** [nextgenhealthcare/connect#5565] [Q&A]
I am writing a channel that creates a DB connection, and then does some processing based on the results of a query. I am running in to issues however when the query is returning NULL, which would be the case if a site did not exist yet in my 'sites' table. How would I handle that possibility?

I have tried messing with size() and .next() but cannot seem to get it figured out.

    var sql = "SELECT MAX(site_score) FROM site WHERE name= '" + name + "' AND address= '" + address + "'"
    var result = dbConn.executeCachedQuery(sql)
    
    if(result.size() == 0){
        return
    }

**Comment by pacmano1** [comment]
Over time I moved most stuff to _always_ return a result:
```SQL
select coalesce (MAX(site_score), 0) FROM site WHERE name = '" + name + "' AND address= '" + address + "'"
```

**Reply by JJWest12** [reply]
So that would always make for it to be at least a 0. How would I do the check in the if statement though? result.getInt('site_score') is giving me and invalid column error

**Reply by pacmano1** [reply]
That is not the only answer of course. It just makes things really consistent as you call database queries.

**Reply by pacmano1** [reply]
```SQL
var query = "select * from urls where is_api_check is false and is_prod is true and url not like '%ap%' and url not like '%api%' and url not like '%lat%'"
var result = executeSharedDBStatement(query, true). // just a code template that calls the DB

logger.info(result.size())  // returns 22

var query = "select * from urls where is_api_check is false and is_prod is true and url =  'ASDF%' and url not like '%api%' and url not like '%plat%'"
var result = executeSharedDBStatement(query, true)
 
logger.info(result.size())  // returns 0
```

**Comment by jonbartels** [comment]
Typical code would do `while(result.next())`. The result set is of an unknown size so JDBC doesn't have to read many rows from DB and into memory if you don't actually need the whole set.

You can also do `if(!result.next())` or `if(!result.first())` (See https://docs.oracle.com/javase/7/docs/api/java/sql/ResultSet.html#first()). If `next()` is false then there are no more rows. If `first()` is false then there is no first row and no rows returned.
